pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_grant_in_txn failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_grant_in_txn [failed](https://tanzanite.cluster.engflow.com/invocation/3e738b04-e5e2-484b-a3cd-c5a7464528d9) on master @ [e747f6e6857a19d6048cb184b0f55c52cb8a6390](https://github.com/cockroachdb/cockroach/commits/e747f6e6857a19d6048cb184b0f55c52cb8a6390):


```
=== RUN   TestTenantLogic_grant_in_txn
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantLogic_grant_in_txn396699939
    test_log_scope.go:81: use -show-logs to present logs inline
[10:13:08] setting distsql_workmem='33042B';
[10:13:11] setting distsql_workmem='33042B';
[10:13:24] --- done: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn with config 3node-tenant: 1 tests, 0 failures
    logic.go:4005: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:7: error while processing
    logic.go:4005: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:7: 
        expected success, but found
        (57014) query execution canceled due to statement timeout
        errors.go:369: in init()
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogic_grant_in_txn396699939
--- FAIL: TestTenantLogic_grant_in_txn (95.25s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=20</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_grant_in_txn.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34346

--- a/pkg/sql/logictest/testdata/logic_test/grant_in_txn
+++ b/pkg/sql/logictest/testdata/logic_test/grant_in_txn
@@ -6,60 +6,170 @@ SET statement_timeout = '10s';
 
 statement ok
 CREATE DATABASE IF NOT EXISTS db1;
+
+statement ok
 CREATE DATABASE IF NOT EXISTS db2;
+
+statement ok
 BEGIN;
+
+statement ok
 CREATE TABLE IF NOT EXISTS db1.t ();
+
+statement ok
 CREATE TABLE IF NOT EXISTS db2.t ();
+
+statement ok
 CREATE USER user1;
+
+statement ok
 CREATE USER user2;
+
+statement ok
 CREATE USER user3;
+
+statement ok
 CREATE USER user4;
+
+statement ok
 CREATE USER user5;
+
+statement ok
 CREATE USER user6;
+
+statement ok
 CREATE USER user7;
+
+statement ok
 CREATE ROLE role1;
+
+statement ok
 CREATE ROLE role2;
+
+statement ok
 CREATE ROLE role3;
+
+statement ok
 CREATE ROLE role4;
+
+statement ok
 CREATE ROLE role5;
+
+statement ok
 CREATE ROLE role6;
+
+statement ok
 CREATE ROLE role7;
+
+statement ok
 CREATE ROLE role8;
+
+statement ok
 GRANT CREATE ON DATABASE db1 TO role1;
+
+statement ok
 GRANT CREATE ON TABLE db1.* TO role1;
+
+statement ok
 GRANT CREATE ON DATABASE db2 TO role1;
+
+statement ok
 GRANT select, insert, delete, update ON TABLE db2.* TO role1;
+
+statement ok
 GRANT role1 TO user5;
+
+statement ok
 GRANT role2 TO user7;
+
+statement ok
 GRANT CREATE ON DATABASE db1 TO role3;
+
+statement ok
 GRANT SELECT, INSERT, DELETE, UPDATE ON TABLE db1.* TO role3;
+
+statement ok
 GRANT ALL ON DATABASE db1 TO role4;
+
+statement ok
 GRANT ALL ON TABLE db1.* TO role4;
+
+statement ok
 GRANT ALL ON DATABASE db1 TO role5;
+
+statement ok
 GRANT ALL ON TABLE db1.* TO role5;
+
+statement ok
 GRANT role5 TO user1;
+
+statement ok
 GRANT CREATE ON DATABASE db2 TO role6;
+
+statement ok
 GRANT SELECT, INSERT, DELETE, UPDATE ON TABLE db2.* TO role6;
+
+statement ok
 GRANT ALL ON DATABASE db2 TO role7;
+
+statement ok
 GRANT ALL ON TABLE db2.* TO role7;
+
+statement ok
 GRANT ALL ON DATABASE db2 TO role8;
+
+statement ok
 GRANT ALL ON TABLE db2.* TO role8;
+
+statement ok
 GRANT admin TO user2;
+
+statement ok
 GRANT admin TO user4;
+
+statement ok
 GRANT admin TO role2;
+
+statement ok
 CREATE ROLE role9;
+
+statement ok
 GRANT role3 TO role9;
+
+statement ok
 GRANT role6 TO role9;
+
+statement ok
 GRANT role9 TO user1;
+
+statement ok
 CREATE ROLE role10;
+
+statement ok
 GRANT role4 TO role10;
+
+statement ok
 GRANT role7 TO role10;
+
+statement ok
 CREATE ROLE role11;
+
+statement ok
 GRANT role5 TO role11;
+
+statement ok
 GRANT role8 TO role11;
+
+statement ok
 GRANT role11 TO user6;
+
+statement ok
 DROP TABLE db1.t;
+
+statement ok
 DROP TABLE db2.t;
+
+statement ok
 COMMIT;
 
 # Ensure that we can inspect information_schema.applicable_roles inside of a

